server/lib/parties/party-service.test.ts

```diff
@@ -0,0 +1,264 @@
+  let client2: InspectableNydusClient
+  let client3: InspectableNydusClient
+  let client4: InspectableNydusClient
+  let client5: InspectableNydusClient
+  let client6: InspectableNydusClient
+  let client7: InspectableNydusClient
+  let client8: InspectableNydusClient
+  let client9: InspectableNydusClient
+
+  let nydus: NydusServer
+  let partyService: PartyService
+  let connector: NydusConnector
+
+  beforeEach(() => {
+    nydus = createFakeNydusServer()
+    const sessionLookup = new RequestSessionLookup()
+    const userSocketsManager = new UserSocketsManager(nydus, sessionLookup, async () => {})
+    const clientSocketsManager = new ClientSocketsManager(nydus, sessionLookup)
+    partyService = new PartyService(nydus, clientSocketsManager, userSocketsManager)
+    connector = new NydusConnector(nydus, sessionLookup)
+
+    client1 = connector.connectClient(user1, USER1_CLIENT_ID)
+    client2 = connector.connectClient(user2, USER2_CLIENT_ID)
+    client3 = connector.connectClient(user3, USER3_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client4 = connector.connectClient(user4, USER4_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client5 = connector.connectClient(user5, USER5_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client6 = connector.connectClient(user6, USER6_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client7 = connector.connectClient(user7, USER7_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client8 = connector.connectClient(user8, USER8_CLIENT_ID)
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    client9 = connector.connectClient(user9, USER9_CLIENT_ID)
+
+    clearTestLogs(nydus)
+  })
+
+  describe('invite', () => {
+    let leader: PartyUser
+    let party: PartyRecord
+
+    describe('when party exists', () => {
+      beforeEach(() => {
+        leader = user1
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user2])
+        partyService.acceptInvite(party.id, user2, USER2_CLIENT_ID)
+      })
+
+      test('should throw if invited by non-leader', () => {
+        const result = () => partyService.invite(user2, USER2_CLIENT_ID, [user3])
+
+        expect(result).toThrow()
+      })
+
+      test('should update the party record', () => {
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user3])
+
+        expect(party.invites).toMatchObject(new Map([[user3.id, user3]]))
+      })
+
+      test('should publish "invite" message to the party path', () => {
+        party = partyService.invite(leader, USER1_CLIENT_ID, [user3])
+
+        expect(nydus.publish).toHaveBeenCalledWith(getPartyPath(party.id), {
+          type: 'invite',
+          invites: [user3],
+        })
+      })
+    })
+
+    describe("when party doesn't exist", () => {
+      beforeEach(() => {
```

Comment:
Not a huge fan of putting these things in a `beforeEach` here. I get that the act of creating a party is common among these tests, but this is actually the code under test for these cases and putting it here obscures that fact. Better to put this in each test case